autoPyTorch/data/utils.py

```diff
@@ -283,8 +463,11 @@ def megabytes(arr: DatasetCompressionInputType) -> float:
 
 def reduce_dataset_size_if_too_large(
     X: DatasetCompressionInputType,
-    memory_allocation: int,
-    methods: List[str] = ['precision'],
+    memory_allocation: Union[int, float],
+    is_classification: bool,
+    random_state: Union[int, np.random.RandomState],
```

Comment:
What should we do for `random_state` in the future?
Should we stick to `np.random.RandomState` or include `int`?
Now everything is `np.random.RandomState`, so I think it is easier to fix it later if we now stick to `np.random.RandomState`

Reply:
When the user can instantiate the class or call the method, we should allow both. I think that's similar to sklearn. Internally we can stick with `np.random.RandomState`. What do you think?

Reply:
That makes sense.